src/api/Controllers/LayerController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using BDMS.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace BDMS.Controllers;
+
+[ApiController]
+[Route("api/v{version:apiVersion}/[controller]")]
+public class LayerController : ControllerBase
+{
+    private readonly BdmsContext context;
+    private readonly ILogger logger;
+
+    public LayerController(BdmsContext context, ILogger<LayerController> logger)
+    {
+        this.context = context;
+        this.logger = logger;
+    }
+
+    /// <summary>
+    /// Asynchronously gets the <see cref="Layer"/>s, optionally filtered by <paramref name="profileId"/>.
+    /// </summary>
+    /// <param name="profileId">The id of the stratigraphy containing the layers to get.</param>
+    [HttpGet]
+    public async Task<ActionResult<IEnumerable<Layer>>> GetAsync(int? profileId = null)
```

Comment:
PP: Diese Methode hätte ich in zwei Methoden `GetAllAsync` und `GetByProfileIdAsync` aufgeteilt, da `GetByIdAsync` auch eine separate Methode darstellt und so die Zuständigkeiten klar sind.